src/core/asyncUtils.ts

```diff
@@ -10,8 +10,8 @@ import {
 import { createTimeoutController } from '../helpers/createTimeoutController'
 import { TimeoutError } from '../helpers/error'
 
-const DEFAULT_INTERVAL = 50
 const DEFAULT_TIMEOUT = 1000
+const DEFAULT_INTERVAL = 50
 
 function asyncUtils(act: Act, addResolver: (callback: () => void) => void): AsyncUtils {
   const wait = async (callback: () => boolean | void, { interval, timeout }: WaitOptions) => {
```

Comment:
using `WaitOptions` is a bit problematic here as the `undefined` has actually been resolved by this point for each of the actual util wrappers.

I've been playing with introducing a `ResolvedWaitOptions` type here to remove it so we don't have to do any casting below:
```suggestion
type ResolvedWaitOptions = { [T in keyof WaitOptions]-?: WaitOptions[T] }

function asyncUtils(act: Act, addResolver: (callback: () => void) => void): AsyncUtils {
  const wait = async (callback: () => boolean | void, { interval, timeout }: ResolvedWaitOptions) => {
```

There's probably a utility type somewhere (built in or `type-fest` that would give a nice generic wrapper for this.  This was just playing around to see if it was possible.

Reply:
Yep, you can totally just replace `WaitOptions` with `Required<WaitOptions>` and remove the casts

```suggestion
function asyncUtils(act: Act, addResolver: (callback: () => void) => void): AsyncUtils {
  const wait = async (callback: () => boolean | void, { interval, timeout }: Required<WaitOptions>) => {
```

Reply:
👍